peps/pep-0733.rst

```diff
@@ -0,0 +1,670 @@
+  and perform operations on them
+* Introspect objects, including types, instances, and functions
+* Raise and handle exceptions
+* Import modules
+* Access to Python's OS interface
+
+The following sections look at the unique requirements of various stakeholders.
+
+Extension Writers
+-----------------
+
+Extension writers are the traditional users of the C API. Their requirements
+are the common actions listed above.
+
+Authors of Embedded Python Applications
+---------------------------------------
+
+Applications with an embedded Python interpreter. Examples are
+`Blender <https://docs.blender.org/api/current/info_overview.html>`__ and
+`OBS <https://obsproject.com/wiki/Getting-Started-With-OBS-Scripting>`__.
+
+They need to be able to:
+
+* Configure the interpreter (import paths, inittab, ``sys.argv``, etc.).
+* Interact with the execution model and program lifetime, including
+  clean interpreter shutdown and restart.
+* Represent complex data models in a way Python can use without
+  having to create deep copies.
+* Provide and import frozen modules.
+* Run and manage multiple independent interpreters (in particular, when
+  embedded in a library that wants to avoid global effects).
+
+Python Implementations
+----------------------
+
+Python implementations such as
+`CPython <https://www.python.org>`__,
+`PyPy <https://www.pypy.org>`__,
+`GraalPy <https://www.graalvm.org/python/>`__,
+`IronPython <https://ironpython.net>`__,
+`RustPython <https://github.com/RustPython/RustPython>`__,
+`MicroPython <https://micropython.org>`__,
+and `Jython <https://www.jython.org>`__), may take
+very different approaches for the implementation of
+different subsystems. They need:
+
+* The API to be abstract and hide implementation details.
+* A specification of the API, ideally with a test suite
+  that ensures compatibility.
+* It would be nice to have an ABI that can be shared
+  across Python implementations.
+
+Alternative APIs and Binding Generators
+---------------------------------------
+
+There are several projects that implement alternatives to the
+C API, which offer extension users advantanges over programming
+directly with the C API. These APIs are implemented with the
+C API, and in some cases by using CPython internals.
+
+There are also libraries that create bindings between Python and
+other object models, paradigms or languages.
+
+There is overlap between these categories: binding generators
+usually provide alternative APIs, and vice versa.
+
+Examples are
+`Cython <https://cython.org>`__,
+`cffi <https://cffi.readthedocs.io/>`__,
+`pybind11 <https://pybind11.readthedocs.io/en/stable/>`__ and
+`nanobind <https://github.com/wjakob/nanobind>`__ for C++,
+`PyO3 <https://github.com/PyO3/pyo3>`__ for Rust,
+`PySide <https://pypi.org/project/PySide/>`__ for Qt,
+`PyGObject <https://pygobject.readthedocs.io/en/latest/>`__ for GTK,
+`Pygolo <https://gitlab.com/pygolo/py>`__ for Go,
+`PyJNIus <https://github.com/kivy/pyjnius/>`__ for Java,
```

Comment:
It is awkward to ask someone to review a request and then endorse a competitor.   The contribution to the Python CAPI review on which portions of this PEP was based was performed Karl Nelson (aka Thrameos) who write for JPype which is Java binding for Python for window/unix not PyJNIus which is the Java binding for Andriod.

Reply:
Beyond this minor quip, I have reviewed the much of this PEP and whole hardly agree with the goal and the accuracy of the stated problems. I would plus up many of the specific subsections as problems that i frequently have encountered if it didn't clutter up the review process.

Reply:
Nothing in the PEP constitutes an "endorsement" of any of the examples listed here. This is not a list of recommendations, just an illustration that there's a thriving ecosystem of tools in this category (which is important, because most people wouldn't be able to name a single one, let alone recognise that it's an important type of project for us to support).

Reply:
Seems worth to also mention [JPype](https://jpype.readthedocs.io/en/latest/) though. I'll send a suggestion for that.